HIP/hip-100.md

```diff
@@ -0,0 +1,479 @@
+---
+hip: 100
+title: Signature and Wallet Providers
+author: Daniel Akhteorv <daniel@launchbadge.com>
+type: Standards Track
+category: Application
+needs-council-approval: Yes
+status: Draft
+created: 2022-01-20
+discussions-to: TBD
+updated: 2022-01-20
+---
+
+## **Abstract**
+
+Ethereum like signature and wallet providers with the Hedera SDKs
+
+## **Motivation**
+
+Cryptocurrency users are more used to Ethereum's signature and wallet providers instead
+of direclty hardcoding private keys into their projects. Supporting third party signature
+and wallet provider would be better for security and user experience.
+
+## **Specification**
+
+<details>
+<summary>abstract class Signer</summary>
+  
+### Methods
+
+##### `async signMessage` ( `messages`: `List` < `bytes` > ): `List` < `List` < `SignerSignature` > >
+
+Sign a list of messages
+
+**NOTE**: Each element in the outer list of the result is all the signatures for the
+at the same index.
+
```

Comment:
Presume these are arbitrary data payloads, not necessarily transactions or even "messages" ?
Suggest renaming to `signData` or just `sign` since the data could be anything and we have an explicit `signTransaction` method.